ui/src/components/PopoverTrigger/PopoverTrigger.jsx

```diff
@@ -6,63 +6,94 @@ import Popover from 'react-bootstrap/Popover';
-                    console.log(err)
-                    return Promise.reject(err.response)
-                }
+        addCity();
+    }
+
+    function handleUnstar() {
+        // Makes POST request to remove city from DB and changes "starred" state variable to be false, causes page to re-render
+        setLoading(true);
+        const removeCity = async () => {
+            try {
+                const res = await axios.post(`${config.API_BASE_URL}/cities/remove`, {
+                    city: name
+                })
+            } catch (err) {
+                alert(err)
+                return Promise.reject(err.response)
             }
 
-            removeCity()
+            setLoading(false);
+            setStarred(false);
         }
-    }, [starred])
 
-    return (
-        <OverlayTrigger key={name} rootClose trigger="click" placement="right" overlay={
-            <Popover className="popover" id="popover-basic" key="state">
-                <Popover.Header >
-                    <h4 className="popover-title">{name}</h4>
-                    {starred ?
-                        <BsStarFill className="popover-star" onClick={() => handleStar()} /> :
-                        <BsStar className="popover-star" onClick={() => handleStar()} />}
-                </Popover.Header>
+        removeCity();
+    }
+
+    if (starred) {
+        return (
+
+            <OverlayTrigger key={name} rootClose trigger="click" placement="right" overlay={
+                <Popover className="popover" id="popover-basic" key="state">
+                    <Popover.Header >
+                        <h4 className="popover-title">{name}</h4>
+                        <BsStarFill className="popover-star" onClick={() => handleUnstar()} />
+                    </Popover.Header>
+
+                    <Popover.Body>Api call using name as search param here</Popover.Body>
+                </Popover>}>
+            <g
+                fill="#1266F1"
+                stroke="#fff"
+                strokeWidth="1"
+                strokeLinecap="round"
+                strokeLinejoin="round"
+                transform="translate(-12, -24)"
+            >
+                <circle cx="12" cy="10" r="3" />
+                <path d={MARKER_SVG_OUTLINE} />
+            </g>
+            </OverlayTrigger>
+        )
+    }
+    else {
+        return (
+            <OverlayTrigger key={name} rootClose trigger="click" placement="right" overlay={
+                <Popover className="popover" id="popover-basic" key="state">
+                    <Popover.Header >
+                        <h4 className="popover-title">{name}</h4>
+                        <BsStar className="popover-star" onClick={() => handleStar()} />
+                    </Popover.Header>
+
+                    <Popover.Body>Api call using name as search param here</Popover.Body>
+                </Popover>}>
+                <circle r={3.5} fill="#1266F1" stroke="#fff" strokeWidth={1} />
```

Comment:
i notice you use this color in a couple places - can you add it into a JS class file - you can have one called "styling" where you define any common JSX styling used across the app (this can be done in a  followup PR)